### DISCUSSION TITLE
Extending v6 Context

### DISCUSSION THREAD

**Discussion by gautamsi** [keystonejs/keystone#5647] [Feature requests]
I am working on a scenario that when there is `sudo` work, I want to assign updatedBy and createdBy to a system user.

most convenient way I can think of is by extending the context and add few things
* `context.system()` similar to `context.sudo()` - (one can also do multiple like `context.guest()` or `context.service()` etc.)
* populate system user id in cache somewhere - need to figure out

OR

have some configuration to automatically do this on `context.sudo()`

Next I may ask on how to associate `on behalf of`, like to capture in which user request it was done by `sudo`

thoughts?

**Comment by bladey** [comment]
Thanks for the suggestion @gautamsi.

**Reply by gautamsi** [reply]
this one seems to be in progress with #6302 

hope this lands soon.
